projects/carba-detect/sawtooth.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+#-------------------------------------------------------------------------------
+# Name:        MCP4921_sawtooth.py
+# Purpose:     Output a sawtooth waveform
+#-------------------------------------------------------------------------------
+
+import spidev
+from time import sleep
+import math
+
+spi_max_speed = 4 * 1000000 # 4 MHz
+V_ref = 5 # 5V 
+max_value = (2**12)-1 # 12 bits for the MCP 4921
+CE = 1 # CE0 or CE1, select SPI device on bus
+
+# setup and open an SPI channel
+spi = spidev.SpiDev()
+spi.open(0,CE)
+spi.max_speed_hz = spi_max_speed
+
+freq = 90
+period = float(1/freq)
+time_sleep =float((period/2)/max_value)
+
+amplitude = 2.39
```

Comment:
by modifying the amplitude var, you can increase/decrease the amplitude of the triangle wave. The 2.39 was the last value we put on the oscilloscope